src/coreclr/jit/emitxarch.cpp

```diff
@@ -1700,13 +1863,45 @@ bool emitter::HasMaskReg(const instrDesc* id) const
 bool IsExtendedReg(regNumber reg)
 {
 #ifdef TARGET_AMD64
+    // TODO-XArch-apx: extend the gpr test, extended gprs should be from r8 to r31 after apx.
     return ((reg >= REG_R8) && (reg <= REG_R15)) || ((reg >= REG_XMM8) && (reg <= REG_XMM31));
 #else
     // X86 JIT operates in 32-bit mode and hence extended reg are not available.
     return false;
 #endif
 }
 
+bool emitter::IsExtendedGPReg(regNumber reg) const
+{
+// TODO-XArch-apx:
+// Consider merge this method into IsExtendedReg(regNumber reg)
+#ifdef TARGET_AMD64
+    if (reg > REG_STK)
+    {
+        // not an actual reg
+        return false;
+    }
+
+// TODO-apx: It would be better to have stress mode on LSRA to forcely allocate EGPRs,
+//           instead of stressing here.
+#if defined(DEBUG)
+    if (emitComp->DoJitStressRex2Encoding())
+    {
+        return true;
+    }
+#endif // DEBUG
+
+    if (UseRex2Encoding())
```

Comment:
Is this check actually necessary?

`IsExtendedReg` freely returns `true` for `XMM16-XMM31` without a corresponding EVEX check. So I would have expected this can simply be `(reg >= REG_R16) && (reg <= REG_R31)` -- with a temporary path doing `||  (DoJitStressRex2Encoding() && (reg >= REG_RAX) && (reg <= REG_R15))` until LSRA is properly updated, but as per the above I'm not sure that's really needed

Reply:
`IsExtendedGPReg` is only used in `HasExtendedGPReg`, which is then used in `TakesRex2Prefix`, I think it should be fine to leave a simple check like `(reg >= REG_R16) && (reg <= REG_R31)` there. If we stress REX2 encoding, `TakesRex2Prefix` will return true by `DoJitStressRex2Encoding` check and never reach the register check.

For now, since the EGPR definition is still missing, I will simply return false there, and come back when EGPRs are defined.  I will document this with an issue. (I'd expect this could be addressed within the PR for register allocator